### PR TITLE
feat(kv-loader): implement loader component

### PR DESCRIPTION
**Pull request by ricardoarasilva**

### CODE REVIEW

File: packages/react-ui-components/src/stories/Components/Loader.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import React from 'react';
+import { KvLoader } from '../../components';
+
+export default {
+	title: "Components/Loader",
+	component: KvLoader,
+	argTypes: {
+		state: {
+			control: { type: "select" },
+			options: Object.values(KvLoader)
```
Comment (pedrorebelocosta) [inline packages/react-ui-components/src/stories/Components/Loader.stories.tsx]: double check this, I think you're getting the keys of the component in itself and not the enum

File: packages/ui-components/.gitignore [hunk 2]
```diff
@@ -1,6 +1,6 @@
 dist/
 www/
-loader/
+./loader/
```
Comment (pedrorebelocosta) [inline packages/ui-components/.gitignore:3]: why the change to a relative path?

File: packages/ui-components/src/components/loader/loader.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+	/** (optional) If `true` the loader is overlay */
+	@Prop({ reflect: true }) hasOverlay: boolean = false;
+
+	render() {
+		return (
+			<Host>
+				{
+					this.isLoading && <div class="loader-container">
+						{
+							this.hasOverlay && <div class="overlay"></div>
```
Comment (pedrorebelocosta) [inline packages/ui-components/src/components/loader/loader.tsx:20]: one liner please

File: packages/ui-components/src/components/switch-button/test/switch-button.e2e.ts [hunk 4]
```diff
@@ -1,110 +0,0 @@
-import { E2EElement, E2EPage, EventSpy, newE2EPage } from '@stencil/core/testing';
```
Comment (pedrorebelocosta) [inline packages/ui-components/src/components/switch-button/test/switch-button.e2e.ts:1]: why was this file deleted/moved?

File: packages/react-ui-components/src/stories/Components/Loader.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+import { KvLoader } from '../../components';
+
+KvLoader.displayName = 'KvLoader';
+
+export default {
+	title: "Components/Loader",
+	component: 'kv-loader',
+	argTypes: {
+		isLoading: { type: "boolean" },
+		hasOverlay: { type:"noolean" }
```
Comment (ricardogoncalves89) [inline packages/react-ui-components/src/stories/Components/Loader.stories.tsx:12]: typo `boolean`

File: packages/ui-components/loader/cdn.js [hunk 6]
```diff
@@ -0,0 +1,3 @@
+
```
Comment (ricardogoncalves89) [inline packages/ui-components/loader/cdn.js:1]: Delete this folder (`packages/ui-components/loader`) that you don't need

File: packages/ui-components/src/components/loader/loader.scss [hunk 7]
```diff
@@ -0,0 +1,45 @@
+@import '../../assets/styles/globals';
+
+$pk-loader-overlay-color: #b9c7c9;
```
Comment (ricardogoncalves89) [inline packages/ui-components/src/components/loader/loader.scss:3]:
```suggestion
$loader-overlay-color: var(--loader-overlay-color, kv-color('neutral-4'));
$loader-overlay-opacity: var(--loader-overlay-opacity, 0.4);
$loader-background-color: var(--loader-background-color, kv-color('neutral-1'));
$loader-spinner-color: var(--loader-spinner-color, kv-color('primary'));
```

1- declare the vars at the beginning of the file it's easier to identify all colors used.  
2- with this implementation it's possible to easily change the colors and opacity from outside the lib.
3- we should only use colors in our [color palette](https://kelvininc.github.io/ui-components/?path=/docs/foundation-colors--page).

File: packages/ui-components/src/components/loader/loader.scss [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+	height: 100%;
+	position: absolute;
+	top: 0;
+	left: 0;
+	z-index: 999;
+
+	.overlay {
+		width: 100%;
+		height: 100%;
+		opacity: var(--loader-overlay-opacity, 0.4);
```
Comment (ricardogoncalves89) [inline packages/ui-components/src/components/loader/loader.scss:17]: Adjust to new vars `$loader-overlay-opacity`

File: packages/ui-components/src/components/loader/loader.scss [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+		background-color: var(--loader-overlay-color, $pk-loader-overlay-color);
```
Comment (ricardogoncalves89) [inline packages/ui-components/src/components/loader/loader.scss:18]: same

File: packages/ui-components/src/components/loader/loader.scss [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+	}
+
+	.loader {
+		height: 0;
+		width: 0;
+		padding: 15px;
+		border: 6px solid var(--loader-background-color, $pk-loader-background-color);
```
Comment (ricardogoncalves89) [inline packages/ui-components/src/components/loader/loader.scss:25]: same

File: packages/ui-components/src/components/loader/loader.scss [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+		border-right-color: var(--loader-spinner-color, kv-color('neutral-4'));
```
Comment (ricardogoncalves89) [inline packages/ui-components/src/components/loader/loader.scss:26]: same

File: packages/ui-components/src/components/loader/readme.md [hunk 12]
```diff
@@ -0,0 +1,18 @@
+# kv-loader
```
Comment (ricardogoncalves89) [inline packages/ui-components/src/components/loader/readme.md:1]:
```suggestion
# *<kv-loader>*
```

and before you committing the code, you should run `npm run build:packages` to be able to have this Readme fully updated

File: packages/ui-components/src/components/loader/usage/react.md [hunk 13]
```diff
@@ -0,0 +1,16 @@
+```tsx
+import React from 'react';
+
+import { KvSwitchButton } from '@kelvininc/react-ui-components';
```
Comment (ricardogoncalves89) [inline packages/ui-components/src/components/loader/usage/react.md:4]: `KvLoader`

File: packages/ui-components/src/components/loader/usage/react.md [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+
+export const SwitchButtonExample: React.FC = () => (
```
Comment (ricardogoncalves89) [inline packages/ui-components/src/components/loader/usage/react.md:6]: LoaderExample

File: packages/ui-components/src/components/loader/usage/react.md [hunk 15]
```diff
@@ -0,0 +1,16 @@
...
+  <>
+    {/*-- Default --*/}
+	<kv-loader is-loading></kv-loader>
```
Comment (ricardogoncalves89) [inline packages/ui-components/src/components/loader/usage/react.md:9]:
```suggestion
	<KvLoader is-loading/>
```

File: packages/ui-components/src/components/loader/usage/react.md [hunk 16]
```diff
@@ -0,0 +1,16 @@
...
+
+	{/*-- Has Overlay --*/}
+	<kv-loader is-loading has-overlay></kv-loader>
```
Comment (ricardogoncalves89) [inline packages/ui-components/src/components/loader/usage/react.md:12]: the same as above

**Review by ricardogoncalves89 — CHANGES_REQUESTED** [review]
Before starting, rebase the branch with dev and reinstall all the dependencies

File: packages/ui-components/.gitignore (hunk 2 above)
Comment (miguelpinto98) [reply packages/ui-components/.gitignore:3]: Stenciljs generate a loader directory, so I assume that change is to git get the changes on the kv-loader. Probably loader isn't a good name, since stenciljs use it.

File: packages/ui-components/loader/cdn.js (hunk 6 above)
Comment (miguelpinto98) [reply packages/ui-components/loader/cdn.js:1]: This directory should be under the gitignore file.

File: packages/react-ui-components/src/stories/Components/Loader.stories.tsx (hunk 1 above)
Comment (ricardoarasilva) [reply packages/react-ui-components/src/stories/Components/Loader.stories.tsx]: Hi @pedrorebelocosta it was fixed yesterday, see the raw of this file https://raw.githubusercontent.com/kelvininc/ui-components/loader-component/packages/react-ui-components/src/stories/Components/Loader.stories.tsx maybe the GitHub is with same bug because doesn't show the correct content here.

File: packages/ui-components/src/components/switch-button/test/switch-button.e2e.ts [hunk 17]
```diff
@@ -26,6 +26,8 @@ describe('Switch Button (end-to-end)', () => {
 
 				const buttonElement = await page.find('kv-switch-button >>> .switch-button');
 				await buttonElement.click()
+
+				await page.waitForTimeout(300)
```
Comment (miguelpinto98) [inline packages/ui-components/src/components/switch-button/test/switch-button.e2e.ts:30]: The linter don't catch the ; at the end?

Comment (ricardoarasilva) [reply packages/ui-components/src/components/switch-button/test/switch-button.e2e.ts:30]: @miguelpinto98 I recreate this system in this branch cause I was deleted this, so, the content is exact the same of dev branch and I think that I can't fix anything here.

File: packages/ui-components/.gitignore (hunk 2 above)
Comment (miguelpinto98) [reply packages/ui-components/.gitignore]: @ricardoarasilva remove the `.` at the beginning

File: packages/ui-components/src/components/switch-button/test/switch-button.e2e.ts (hunk 17 above)
Comment (miguelpinto98) [reply packages/ui-components/src/components/switch-button/test/switch-button.e2e.ts:30]: you can add the trailing comma that are missing in the file... In other hand, the `waitForTimeout` is added in this change.

File: packages/react-ui-components/src/stories/Components/Loader.stories.tsx [hunk 18]
```diff
@@ -0,0 +1,31 @@
...
+
+export default {
+	title: "Components/Loader",
+	component: 'kv-loader',
+	argTypes: {
+		isLoading: { type: "boolean" },
+		hasOverlay: { type: "boolean" }
+	},
+	parameters: {
+		notes: require('../../../../ui-components/src/components/loader/readme.md')
```
Comment (ricardogoncalves89) [inline packages/react-ui-components/src/stories/Components/Loader.stories.tsx:15]: please rebase with dev branch and update to `require('@ui-notes/loader/readme.md')`

File: packages/ui-components/src/components/loader/loader.scss [hunk 19]
```diff
@@ -0,0 +1,47 @@
+@import '../../assets/styles/globals';
+
+$loader-overlay-color: var(--loader-overlay-color, kv-color('neutral-4'));
```
Comment (ricardogoncalves89) [inline packages/ui-components/src/components/loader/loader.scss:3]: Sorry I forgot to tell you to put all styles inside `:host { ... }`

**Review by ricardogoncalves89 — CHANGES_REQUESTED** [review]

File: packages/ui-components/.gitignore (hunk 2 above)
Comment (ricardogoncalves89) [reply packages/ui-components/.gitignore]: @ricardoarasilva after make the change you need to mark as resolved

File: packages/react-ui-components/src/stories/Components/Loader.stories.tsx (hunk 5 above)
Comment (ricardogoncalves89) [reply packages/react-ui-components/src/stories/Components/Loader.stories.tsx]: @ricardoarasilva after make the change you need to mark as resolved

File: packages/ui-components/src/components/loader/loader.scss (hunk 7 above)
Comment (ricardogoncalves89) [reply packages/ui-components/src/components/loader/loader.scss]: @ricardoarasilva after make the change you need to mark as resolved

**Review by ricardogoncalves89 — CHANGES_REQUESTED** [review]
If you already rebase with dev please:
1- Delete the folders`angular-ui-components/src/generated` this folder don't exists anymore

2 - the folder `react-ui-components/src/components/stencil-generated` and file `components.d.ts` should be ignored by .gitignore

**Review by ricardogoncalves89 — APPROVED** [review]

**Review by pedrorebelocosta — APPROVED** [review]

**Review by miguelpinto98 — APPROVED** [review]
🎸
